Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/negotiation/services/NegotiationService.kt

```diff
@@ -67,11 +70,25 @@ class NegotiationService constructor(val agreementsRepository: AgreementsReposit
                                     CreatedDate = Date(),
                                     MovedToBlockChain = false,
                                     AgreementImageURL = createAgreementRequest.ImageURL
-                                    ).apply { partyA = userA}
-        nAgreement = nAgreement.apply { partyB = userB }
+                                    )
+        nAgreement = nAgreement.apply { users.add(userA)}
+        nAgreement = nAgreement.apply { users.add(userB)}
+
+        val tempA = agreementsRepository.getAllByUsersContaining(userA) ?:
+        mutableSetOf()
+
+        val tempB = agreementsRepository.getAllByUsersContaining(userB) ?:
+        mutableSetOf()
```

Comment:
This seems redundant. If these values are set, wouldn't you already get this info from the userRepository?

Reply:
No its due to the lazy initialization. I kept on getting a lazy exception as it tries to access the agreements from user which was not fetched. I needed to do this such that I can add the new agreement to the agreement set